src/compiler/config/test/validate-output-dist.spec.ts

```diff
@@ -41,7 +41,7 @@ describe('validateDistOutputTarget', () => {
         empty: false,
         isBrowserBuild: true,
         legacyLoaderFile: path.join(rootDir, 'my-dist', 'my-build', 'testing.js'),
-        polyfills: true,
+        polyfills: false,
```

Comment:
The changes from `undefined` to `false` here make sense to me, but the value changes from `true` to `false` less so. Can you help me understand why this is happening?

Reply:
[This comment gives some context](https://github.com/ionic-team/stencil/pull/4815#discussion_r1330579976), basically I changed it to ensure that the value on the generated output targets matches what it set on the `"dist"` output target (the logic was more confusing before, if `.polyfills` was `undefined` on the `dist` output target then it would end up always being `true` on the `dist-lazy` one, which doesn't seem right?)

Reply:
I agree it seems like they should be in sync

Reply:
mm yeah it's a bit complicated, I guess previously we actually had three different states we treated, basically

1. `dist.polyfills` is `true`, then `dist-lazy` should be true
2. `dist.polyfills` is `false`, then `dist-lazy` should be false
3. `dist.polyfills` is `undefined`, then `dist-lazy` should be `true`

I think the issue is that right now the `undefined` case isn't distinguished from the `false` case, I'll make the logic here check the original output target instead of the validated one (return value of `validateOutputTargetDist`)